enterprise/api/v1/views.py

```diff
@@ -510,3 +515,94 @@ class EnterpriseCustomerReportingConfigurationViewSet(EnterpriseReadOnlyModelVie
     )
     filter_fields = FIELDS
     ordering_fields = FIELDS
+
+
+class CouponCodesView(APIView):
+    """
+    API to request coupon codes.
+    """
+    permission_classes = (permissions.IsAuthenticated,)
+    authentication_classes = (JwtAuthentication, BearerAuthentication, SessionAuthentication,)
+    throttle_classes = (ServiceUserThrottle,)
+
+    REQUIRED_PARAM_EMAIL = 'email'
+    REQUIRED_PARAM_ENTERPRISE_NAME = 'enterprise_name'
+    OPTIONAL_PARAM_NUMBER_OF_CODES = 'number_of_codes'
+
+    MISSING_REQUIRED_PARAMS_MSG = "Some required parameter(s) missing: {}"
+
+    def get_required_query_params(self, request):
+        """
+        Gets ``email``, ``enterprise_name``, and ``number_of_codes``,
+        which are the relevant parameters for this API endpoint.
+
+        :param request: The request to this endpoint.
+        :return: The ``email``, ``enterprise_name``, and ``number_of_codes`` from the request.
+        """
+        email = get_request_value(request, self.REQUIRED_PARAM_EMAIL, '')
+        enterprise_name = get_request_value(request, self.REQUIRED_PARAM_ENTERPRISE_NAME, '')
+        number_of_codes = get_request_value(request, self.OPTIONAL_PARAM_NUMBER_OF_CODES, '')
+        if not (email and enterprise_name):
+            raise CodesAPIRequestError(
+                self.get_missing_params_message([
+                    ("'email'", bool(email)),
```

Comment:
Rather than the hardcoded string I'd recommend using REQUIRED_PARAM_EMAIL, REQUIRED_PARAM_ENTERPRISE_NAME you set above